src/figuresClasses.ts

```diff
@@ -2,10 +1,83 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a < 1 || this.b < 1 || this.c < 1) {
+      throw new Error('Sides can`t be less than 1');
+    }
+
+    const p = this.a + this.b + this.c;
+
+    const maxValue = Math.max(this.a, this.b, this.c);
+
+    if (maxValue >= p - maxValue) {
+      throw new Error('It is not a triangle');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius < 1) {
+      throw new Error('Radius can`t be less than 1');
```

Comment:
[CODE STYLE] - The error message can be improved for clarity. Suggested message: 'Circle radius must be greater than 0.'